src/Humanizer/Localisation/NumberToWords/SpanishNumberToWordsConverter.cs

```diff
@@ -1,193 +1,383 @@
 
+            if (number == long.MinValue)
+            {
+                return
+                    "menos nueve trillones doscientos veintitrés mil trescientos setenta y dos billones treinta y seis mil " +
+                    "ochocientos cincuenta y cuatro millones setecientos setenta y cinco mil ochocientos ocho";
+            }
+
             if (number < 0)
             {
-                return string.Format("menos {0}", Convert(Math.Abs(number)));
+                return $"menos {Convert(-number)}";
             }
 
-            var parts = new List<string>();
+            wordBuilder.Add(ConvertGreaterThanMillion(number, out var remainder));
+            wordBuilder.Add(ConvertThousands(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertHundreds(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertUnits(remainder, gender, wordForm));
+
+            return BuildWord(wordBuilder);
+        }
+
+        public override string ConvertToOrdinal(int number, GrammaticalGender gender)
+        {
+            return ConvertToOrdinal(number, gender, WordForm.Normal);
+        }
+
+        public override string ConvertToOrdinal(int number, GrammaticalGender gender, WordForm wordForm)
+        {
+            List<string> wordBuilder = new();
 
-            if ((number / 1000000000) > 0)
+            if (number == 0 || number == int.MinValue)
             {
-                parts.Add(number / 1000000000 == 1
-                    ? "mil millones"
-                    : string.Format("{0} mil millones", Convert(number / 1000000000)));
+                return "cero";
+            }
 
-                number %= 1000000000;
+            if (number < 0)
+            {
+                return ConvertToOrdinal(Math.Abs(number), gender);
             }
 
-            if ((number / 1000000) > 0)
+            if (IsRoundBillion(number))
             {
-                parts.Add(number / 1000000 == 1
-                    ? "un millón"
-                    : string.Format("{0} millones", Convert(number / 1000000)));
+                return ConvertRoundBillionths(number, gender);
+            }
 
-                number %= 1000000;
+            if (IsRoundMillion(number))
+            {
+                return ConvertToOrdinal(number / 1000, gender).Replace("milésim", "millonésim");
             }
 
-            if ((number / 1000) > 0)
+            wordBuilder.Add(ConvertTensAndHunderdsOfThousandths(number, out var remainder, gender));
+            wordBuilder.Add(ConvertThousandths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertHundredths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertTenths(remainder, out remainder, gender));
+            wordBuilder.Add(ConvertOrdinalUnits(remainder, gender, wordForm));
+
+            return BuildWord(wordBuilder);
+        }
+
+        public override string ConvertToTuple(int number)
+        {
+            string[] map = {
```

Comment:
Can this be a static member in the type so we don't allocate on each call?

Reply:
Yes. Maps were static members before I refactor this class.  I thought it was a good idea, which improved readability, to move them close to where they are used. Unfortunately, I didn't consider your concerns. Sorry for that. 
I don't mind promoting them to static members.